dokka-integration-tests/gradle/src/testExampleProjects/kotlin/ExampleProjectsTest.kt

```diff
@@ -0,0 +1,344 @@
+
+                    ExampleProject.Multimodule -> {
+                        shouldHaveTasksWithOutcome(
+                            ":docs:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":childProjectA:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":childProjectB:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":docs:dokkaGeneratePublicationHtml" to UP_TO_DATE,
+                            ":docs:dokkaGenerate" to UP_TO_DATE,
+                        )
+                    }
+
+                    ExampleProject.VersioningMultimodule -> {
+                        shouldHaveTasksWithOutcome(
+                            ":docs:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":childProjectA:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":childProjectB:dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":docs:dokkaGeneratePublicationHtml" to UP_TO_DATE,
+                            ":docs:dokkaGenerate" to UP_TO_DATE,
+                        )
+                    }
+
+                    else -> {
+                        shouldHaveTasksWithOutcome(
+                            ":dokkaGeneratePublicationHtml" to UP_TO_DATE,
+                            ":dokkaGenerateModuleHtml" to UP_TO_DATE,
+                            ":dokkaGenerate" to UP_TO_DATE,
+                        )
+                    }
+                }
+            }
+    }
+
+
+    @ParameterizedTest
+    @ArgumentsSource(TestCaseProvider::class)
+    fun `test configuration cache`(testCase: TestCase) {
+        // delete old configuration cache results and reports, to make sure we can fetch the newest report
+        testCase.project.findFiles {
+            val isCCDir = it.invariantSeparatorsPath.endsWith(".gradle/configuration-cache")
+            val isCCReportDir = it.invariantSeparatorsPath.endsWith("build/reports/configuration-cache")
+            it.isDirectory && (isCCReportDir || isCCDir)
+        }.forEach { it.deleteRecursively() }
+
+        val configCacheRunner: GradleRunner =
+            testCase.project.runner
+                .addArguments(
+                    testCase.dokkaGenerateTask,
+                    "--stacktrace",
+                    "--configuration-cache",
+                )
+
+        //first build should store the configuration cache
+        configCacheRunner.build {
+            output shouldContain "BUILD SUCCESSFUL"
+            output shouldContain "Configuration cache entry stored"
+
+            val ccReport = testCase.project.loadConfigurationCacheReportData().shouldNotBeNull()
+            ccReport.asClue {
+                it.totalProblemCount shouldBe 0
+            }
+        }
+
+        // second build should reuse the configuration cache
+        configCacheRunner.build {
+            output shouldContain "BUILD SUCCESSFUL"
+            output shouldContain "Configuration cache entry reused"
```

Comment:
I think it might make sense to `clean` before rerunning this build; this might catch something like a build output file becoming a CC cache entry. When would this fail otherwise?

Reply:
> this might catch something like a build output file becoming a CC cache entry

Actually, I guess rebuilding after execution phase without cleaning is more likely to catch an issue like that